plugins/core/Standard/qRANSAC_SD/src/qRANSAC_SD.cpp

```diff
@@ -166,10 +181,28 @@ void qRansacSD::doAction()
 	rsdDlg.cylinderCheckBox->setChecked(s_primEnabled[2]);
 	rsdDlg.coneCheckBox->setChecked(s_primEnabled[3]);
 	rsdDlg.torusCheckBox->setChecked(s_primEnabled[4]);
+	rsdDlg.simplifyShapescheckBox->setChecked(s_allowSimplification);
+	rsdDlg.saveLeftOverscheckBox->setChecked(s_createCloudFromLeftOverPoints);
+	rsdDlg.maxSphereRadiuscheckBox->setChecked(s_maxSphereRadiusEnabled);
+	rsdDlg.maxCylinderRadiuscheckBox->setChecked(s_maxCylinderRadiusEnabled);
+	rsdDlg.maxConeAnglecheckBox->setChecked(s_maxConeAngleEnabled);
+	rsdDlg.maxConeRadiuscheckBox->setChecked(s_maxConeRadiusEnabled);
+	rsdDlg.maxSphereRadiusdoubleSpinBox->setValue(s_maxSphereRadius);
+	rsdDlg.maxCylinderRadiusdoubleSpinBox->setValue(s_maxCylinderRadius);
+	rsdDlg.maxConeRadiusdoubleSpinBox->setValue(s_maxConeRadius);
+	rsdDlg.maxConeLengthdoubleSpinBox->setValue(s_maxConeLength);
+	rsdDlg.maxCylinderLengthdoubleSpinBox->setValue(s_maxCylinderLength);
+	rsdDlg.maxConeAngledoubleSpinBox->setValue(s_maxConeAngle_deg);
 
 	if (!rsdDlg.exec())
 		return;
 
+	s_maxSphereRadiusEnabled = rsdDlg.maxSphereRadiuscheckBox->isChecked();
+	s_maxCylinderRadiusEnabled = rsdDlg.maxCylinderRadiuscheckBox->isChecked();
+	s_maxConeRadiusEnabled = rsdDlg.maxConeRadiuscheckBox->isChecked();
+	s_maxConeLengthEnabled = rsdDlg.maxConeLengthcheckBox->isChecked();
+	s_maxCylinderLengthEnabled = rsdDlg.maxCylinderLengthcheckBox->isChecked();
+	s_maxConeAngleEnabled = rsdDlg.maxConeAnglecheckBox->isChecked();
```

Comment:
Maybe I would add an option to toggle least square fitting on inliers set, what do you think? I won't have a specific use case in mind for not using the final round of levmar tought.

Reply:
I actually have considered a final fitting option. Specifically the sphere fit by CC after being found by RANSAC have been much better. (I'm finding spheres the are known metrologic references, and the RANSAC returns were typically a few percentage larger than real, where CCs were within the known characteristic limits for the sphere and my measurement device. 
Only sphere fitting in CC is much slower

Reply:
![image](https://user-images.githubusercontent.com/7495752/84578400-bc4c8180-ad79-11ea-90cd-aba272f434c4.png)
in this image the spheres point clouds and the first sphere primitives are RANSAC returns, and the 2nd sphere primitives are using CC fit sphere functions on the sphere point clouds from RANSAC.
The spheres should be 0.5" and 0.25" +/- 0.005" from the manufacturer

Reply:
Ouch, sphere_0004 fitting by Schnabel's algo is really bad. I though final refinement by least square fitting was enabled by default in the plugin.

Reply:
That is after final refinement, I haven't tried it without it, but I did turn on precision levmar mode to get it that decent. Prior to about a week ago I hadn't ever managed to get anything useful for sphere/cylinder finding with this plugin.